core/src/test/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -321,14 +321,21 @@ public void deleteBlob(String blobName) throws IOException {
 
             @Override
             public void move(String sourceBlob, String targetBlob) throws IOException {
+                // simulate a non-atomic move, since many blob container implementations
+                // will not have an atomic move, and we should be able to handle that
                 maybeIOExceptionOrBlock(targetBlob);
-                super.move(sourceBlob, targetBlob);
+                super.writeBlob(targetBlob, readBlob(sourceBlob), 0L);
```

Comment:
`super.readBlock` instead of `readBlock` to prevent double `maybeIOExceptionOrBlock`.